src/main/java/com/google/starfish/servlets/GetTrendingNotesServlet.java

```diff
@@ -0,0 +1,51 @@
+package com.google.starfish.servlets;  
+
+import java.io.IOException;  
+import java.sql.SQLException;  
+import javax.servlet.ServletException;  
+import javax.servlet.annotation.WebServlet;  
+import javax.servlet.http.HttpServlet;  
+import javax.servlet.http.HttpServletRequest;  
+import javax.servlet.http.HttpServletResponse;  
+import javax.sql.DataSource;
+import com.google.starfish.services.FavoriteNoteService;
+import com.google.starfish.services.FavoriteNoteService.Recency;
+
+/** Servlet that returns trending notes based on a given timespan as a query param */
+@WebServlet("/get-trending-notes")  
+public class GetTrendingNotesServlet extends HttpServlet {  
+
+  private FavoriteNoteService favoriteNoteService = new FavoriteNoteService();
+
+  @Override
+  public void doGet(HttpServletRequest req, HttpServletResponse res) throws ServletException, IOException {
+    DataSource pool = (DataSource) req.getServletContext().getAttribute("my-pool");  
+
+    String timespan = Utils.trimAndLowerCaseString(req.getParameter("timespan"));
+    Recency recency = Utils.findRecencyByString(timespan);
+    // Default recency is all-time
+    if (recency == null) recency = Recency.ALL_TIME;
+    try {
+      Object[][] trendingNotes = null;
+      switch(recency) {
+        case TODAY:
+          trendingNotes = favoriteNoteService.getTrendingNotesBySchoolOrCourse(pool, FavoriteNoteService.Recency.TODAY, null, null);
+          break;
+        case THIS_WEEK:
+          trendingNotes = favoriteNoteService.getTrendingNotesBySchoolOrCourse(pool, FavoriteNoteService.Recency.THIS_WEEK, null, null);
+          break;
+        case THIS_MONTH:
+          trendingNotes = favoriteNoteService.getTrendingNotesBySchoolOrCourse(pool, FavoriteNoteService.Recency.THIS_MONTH, null, null);
+          break;
+        default:
+          // Default to returning trending notes all-time
+          trendingNotes = favoriteNoteService.getTrendingNotesBySchoolOrCourse(pool, FavoriteNoteService.Recency.ALL_TIME, null, null);
```

Comment:
(Here and in file below) actually do you need this switch block anymore?  Couldn't you just say `trendingNotes = favoriteNoteService.getTrendingNotesBySchoolOrCourse(pool, recency, null, null)` and it would accordingly pass the right value for recency?

Reply:
woahhhh you're right haha, turns out there was even more benefit to using that recency enum everywhere!! This really simplifies the code a lot more!